docs/httpproxy.md

```diff
@@ -0,0 +1,833 @@
+
+**Lines 8-9**: Each HTTPProxy must have one or more `routes`, each of which must have a condition to match against (e.g. `prefix: /`) and then one or more `services` which will handle the HTTP traffic.
+
+**Lines 10-12**: The `services` field is an array of named Service & Port combinations that will be used for this HTTPProxy path.
+HTTP traffic will be sent directly to the Endpoints corresponding to the Service.
+
+## Interacting with HTTPProxies
+
+As with all Kubernetes objects, you can use `kubectl` to create, list, describe, edit, and delete HTTPProxy CRDs.
+
+Creating an HTTPProxy:
+
+```bash
+$ kubectl create -f basic.httpproxy.yaml
+httpproxy "basic" created
+```
+
+Listing HTTPProxies:
+
+```bash
+$ kubectl get httpproxy
+NAME      AGE
+basic     24s
+```
+
+Describing HTTPProxy:
+
+```bash
+$ kubectl describe httpprox basic
+Name:         basic
+Namespace:    default
+Labels:       <none>
+API Version:  projectcontour.io/v1alpha1
+Kind:         HTTPProxy
+Metadata:
+  Cluster Name:
+  Creation Timestamp:  2019-07-05T19:26:54Z
+  Resource Version:    19373717
+  Self Link:           /apis/projectcontour.io/v1alpha1/namespaces/default/httpproxy/basic
+  UID:                 6036a9d7-8089-11e8-ab00-f80f4182762e
+Spec:
+  Routes:
+    Conditions:
+      Prefix: /
+    Services:
+      Name:  s1
+      Port:  80
+  Virtualhost:
+    Fqdn:  foo-basic.bar.com
+Events:    <none>
+```
+
+Deleting HTTPProxies:
+
+```bash
+$ kubectl delete httpproxy basic
+httpproxy "basic" deleted
+```
+
+## HTTPProxy API Specification
+
+TODO(dfc) create example directory
+
+There are a number of [working examples](/examples/example-workload/httpproxy) of HTTPProxy objects in the `examples/example-workload` directory.
+
+We will use these examples as a mechanism to describe HTTPProxy API functionality.
+
+### Virtual Host Configuration
+
+#### Fully Qualified Domain Name
+
+Similar to Ingress, HTTPProxy support name-based virtual hosting.
+Name-based virtual hosts use multiple host names with the same IP address.
+
+```
+foo.bar.com --|                 |-> foo.bar.com s1:80
+              | 178.91.123.132  |
+bar.foo.com --|                 |-> bar.foo.com s2:80
+```
+
+Unlike Ingress, HTTPProxy only support a single root domain per HTTPProxy object.
+As an example, this Ingress object:
+
+```yaml
+# ingress-name.yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: name-example
+spec:
+  rules:
+  - host: foo1.bar.com
+    http:
+      paths:
+      - backend:
+          serviceName: s1
+          servicePort: 80
+  - host: bar1.bar.com
+    http:
+      paths:
+      - backend:
+          serviceName: s2
+          servicePort: 80
+```
+
+must be represented by two different HTTPproxy objects:
+
+```yaml
+# httpproxy-name.yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: HTTPProxy
+metadata:
+  name: name-example-foo
+  namespace: default
+spec:
+  virtualhost:
+    fqdn: foo1.bar.com
+  routes:
+    - conditions:
```

Comment:
Conditions on route are singular (e.g. `condition`). Also, they aren't an array for the Route spec since you can only apply a single `Condition` to a specific route: https://github.com/projectcontour/contour/blob/master/apis/projectcontour/v1alpha1/httpproxy.go#L82

Reply:
Resolving this one because this is not the case any more.